sdk/attestation/azure-security-attestation/inc/azure/attestation/attestation_client_models.hpp

```diff
@@ -0,0 +1,122 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+/**
+ * @file
+ * @brief Defines the Azure Attestatoin API types.
+ *
+ */
+
+#pragma once
+
+#include "azure/attestation/dll_import_export.hpp"
+
+#include <azure/core/context.hpp>
+#include <azure/core/http/http.hpp>
+#include <azure/core/nullable.hpp>
+#include <azure/core/paged_response.hpp>
+#include <azure/core/response.hpp>
+#include <iostream>
+#include <memory>
+#include <string>
+#include <unordered_map>
+#include <vector>
+
+namespace Azure { namespace Security { namespace Attestation {
+  class AttestationClient;
+  /**
+   * @brief Contains identity and other basic properties of a Certificate.
+   *
+   */
+  struct AttestationOpenIdMetadata final
```

Comment:
nit. when adding a constructor, we use `class`.  It's a convention I've never liked within the team, but you'll hear about it

Reply:
> I feel fairly strongly that struct == class with all members public.

<3 <3 <3  Where have you been all this time Larry!!! hahahaha   Now we just need to convince @RickWinter @ahsonkhan and @antkmsft about it.  ( I don't remember where's @gearama 's stand about this ... )

Reply:
I think we are on the same page of what's the difference between struct vs class in C++, Victor :)

There's also the following, that Larry said, which largely holds true as well.
> More importantly, model types should always be structs.

The convention was, for model and option types, which almost always just contain public members ("properties"), or option bags, and are otherwise stateless, they should be structs. Everything else, including things with functions, and state ("private fields"), should be classes.

I guess the question here is about "always" vs "usually". I don't see a ctor here, as expected, so struct makes sense, like all other typical model types.

Reply:
I nuked the ctor in the most recent revision because it wasn't needed (I wasn't seeing the default move ctor being called when I debugged the code so I was trying to ensure it was present).

However my general principle holds: Model types (outputs of the API calls) should be structs. They also should be immutable (all fields are const). That last one drives to an interesting problem: Initializing const fields can only be done through a ctor. The default move and copy ctors will likely work well, but that isn't the case for initializing (this is especially true for complicated model types (ten-twenty fields).

I totally agree that model types SHOULD NOT (RFC 2119) have methods. But I think a ctor makes sense if we have immutable model types. Let's take this offline to email rather than GH comments - it's not germane to this particular PR right now, but may be in the future (one of my model types has 17 fields)